agents/hwp_rotation/rotation_agent.py

```diff
@@ -0,0 +1,443 @@
+import sys
+import os
+import argparse
+import time
+
+this_dir = os.path.dirname(__file__)
+sys.path.append(
+        os.path.join(this_dir, 'src'))
+
+import pid_controller as pd
+from socs.agent.pmx import PMX, Command
+
+from ocs import ocs_agent, site_config, client_t
+from ocs.ocs_twisted import TimeoutLock
+
+class RotationAgent:
+    """Agent to control the rotation speed of the CHWP
+
+    Args:
+        kikusui_ip (str): IP address for the Kikusui power supply 
+        kikusui_port (str): Port for the Kikusui power supply
+        pid_ip (str): IP address for the PID controller
+        pid_port (str): Port for the PID controller
+
+    """
+    def __init__(self, agent, kikusui_ip, kikusui_port, pid_ip, pid_port):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.take_data = False
+        self.switching = False
+        self.kikusui_ip = kikusui_ip
+        self.kikusui_port = int(kikusui_port)
+        self.pid_ip = pid_ip
+        self.pid_port = pid_port
+
+        agg_params = {'frame_length': 60}
+        self.agent.register_feed('hwprotation', record = True, agg_params = agg_params)
+
+        try:
+            self.pmx = PMX(tcp_ip = self.kikusui_ip, tcp_port = self.kikusui_port, timeout = 0.5)
+            self.cmd = Command(self.pmx)
+            print('Connected to Kikusui power supply')
+        except:
+            print('Could not establish connection to Kikusui power supply')
+            sys.exit(0)
+
+        try:
+            self.pid = pd.PID(pid_ip = self.pid_ip, pid_port = self.pid_port)
+            print('Connected to PID controller')
+        except:
+            print('Could not establish connection to PID controller')
+            sys.exit(0)
```

Comment:
An exit status of "0" indicates "success". Use another number. There isn't really a convention within OCS for this.

I thought maybe there would be a better way to exit the Agent since the reactor should be running, but I think since this is on the initialization of the Agent object it's not quite running yet and this should be fine. However, there are some issues with the PMX destructor that mean this is going to hit an error sooner if the kikusui is inaccessible.